lib/test_queue/runner.rb

```diff
@@ -162,13 +177,42 @@ def summarize_internal
         puts @failures
       end
 
+      if !relay?
+        unless @discovered_suites.empty?
+          puts
+          puts "The following suites were discovered but were not run:"
+          puts
+
+          @discovered_suites.sort.each do |suite_name, path|
+            puts "#{suite_name} - #{path}"
+          end
+        end
+        unless unassigned_suites.empty?
+          puts
+          puts "The following suites were not discovered but were run anyway:"
+          puts
+          unassigned_suites.sort.each do |suite_name, path|
+            puts "#{suite_name} - #{path}"
+          end
+        end
+        unless misrun_suites.empty?
+          puts
+          puts "The following suites were run on the wrong workers:"
+          puts
+          misrun_suites.each do |suite_name, path, target_host, target_pid, actual_host, actual_pid|
+            puts "#{suite_name} - #{path}: #{actual_host} (#{actual_pid}) - assigned to #{target_host} (#{target_pid})"
+          end
+        end
+      end
+
       puts
 
       @stats.save
 
       summarize
 
       estatus = @completed.inject(0){ |s, worker| s + worker.status.exitstatus }
+      estatus += 1 unless @discovered_suites.empty? && misrun_suites.empty? && unassigned_suites.empty?
```

Comment:
I think `unassigned_suites` will be non-empty on remote masters because `@discovered_suites` will always be empty. Seems like maybe more of the work needs to move into the `if !relay?` block.